### ISSUE TITLE
False Positive: Benchmark-approved parameter-convention cases are still treated as unused formal parameters.

### ISSUE THREAD

**Issue by Carlson-JLQ** [pmd/pmd#6543]
# False Positive: Benchmark-approved parameter-convention cases are still treated as unused formal parameters.

Version
7.22.0

## Checker
- Checker id: `category/java/bestpractices.xml/UnusedFormalParameter`
- Checker description: Detects unused formal parameters in private methods/constructors (or all methods/constructors if configured) unless the parameter name indicates it is intentionally unused.

## Description of the false positive
`diff/output` marks 1 false positives for `category/java/bestpractices.xml/UnusedFormalParameter`. Across these samples, the checker tends to fire on the outer syntax while missing code-level cues that make the pattern harmless, exempted, or at least much weaker than a real bug. The reports read more like over-eager matching than strong bug findings.

### `NegCase6.java`
Source: `output/sonarqube/UnusedMethodParameterCheck/src/main/java/scensct/core/neg/NegCase6.java`

The sample keeps the benchmark's target pattern but expresses it in a slightly non-canonical source form. A human reader would likely classify this as an allowed or intentionally structured pattern rather than a bug. That matches the benchmark's stated intent: Private writeObject method with unused parameter should not be flagged.

```java
// Private writeObject method with unused parameter should not be flagged.
package scensct.core.neg;

import java.io.ObjectOutputStream;
import java.io.IOException;

public class NegCase6 {
    private void writeObject(ObjectOutputStream out) throws IOException {
        // Parameter out is unused, but method is special serialization method.
    }
}
```

## Cause analysis
The rule is documented as follows: Detects unused formal parameters in private methods/constructors (or all methods/constructors if configured) unless the parameter name indicates it is intentionally unused.

 The implementation appears to lean too heavily on surface indicators and not enough on what the code actually does.

That is why these reports read like over-approximation. The code often retains a suspicious signature or control-flow skeleton, but once you read the body, the benchmark's claimed exemption is visible in the source itself.

For `category/java/bestpractices.xml/UnusedFormalParameter`, the practical improvement would be to model those source-level exemptions more explicitly instead of reporting every syntactic near-match.

## References
None known.

**Comment by oowekyala** [comment]
Rename the parameter "unused" or suppress the violation.
